### PR TITLE
[patch] disable health check server when port <0

### PR DESCRIPTION
**Pull request by kevindiu**
… < 0

### CODE REVIEW

File: service/server.go [hunk 1]
```diff
@@ -75,52 +76,55 @@ var (
...
+			Handler: createHealthCheckServiceMux(s.cfg.HealthzPath),
+		}
+		s.hcsrv.SetKeepAlivesEnabled(true)
 	}
 
-	pwt, err := time.ParseDuration(cfg.ProbeWaitTime)
+	s.sddur, err = time.ParseDuration(s.cfg.ShutdownDuration)
 	if err != nil {
-		pwt = time.Second * 3
+		s.sddur = time.Second * 5
```
Comment (WindzCUHK) [inline service/server.go]: can help resolve the conflict?

File: service/server_test.go [hunk 2]
```diff
@@ -404,6 +410,79 @@ func Test_server_ListenAndServe(t *testing.T) {
...
+						srv := &http.Server{
+							Addr:    fmt.Sprintf(":%d", hcSrvPort),
+							Handler: handler,
+						}
+
+						srv.SetKeepAlivesEnabled(true)
+						return srv
+					}(),
+					cfg: config.Server{
+						Port: apiSrvPort,
```
Comment (WindzCUHK) [inline service/server_test.go]:
```suggestion
						Port: apiSrvPort,
						// HealthzPort: hcSrvPort,
```
leave a comment? easier to understand the test case

File: service/server.go (hunk 1 above)
Comment (kevindiu) [reply service/server.go]: resolved

File: service/server_test.go (hunk 2 above)
Comment (kevindiu) [reply service/server_test.go]: added :)

File: service/server.go [hunk 3]
```diff
@@ -75,52 +76,55 @@ var (
...
 
-	return &server{
-		srv:   srv,
-		hcsrv: hcsrv,
-		cfg:   cfg,
-		pwt:   pwt,
-		sddur: dur,
+	s.pwt, err = time.ParseDuration(s.cfg.ProbeWaitTime)
+	if err != nil {
+		s.pwt = time.Second * 3
```
Comment (WindzCUHK) [inline service/server.go]:
```suggestion
		glg.Warn(err)
```
@.@

Comment (kevindiu) [reply service/server.go]: @@

**Comment by WindzCUHK** [comment]
LGTM

**Pull request merged**